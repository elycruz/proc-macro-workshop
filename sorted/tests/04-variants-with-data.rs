// This test is similar to the previous where want to ensure that the macro
// correctly generates an error when the input `enum` is out of order, but this
// time we're working with an `enum` that has a data payload on each variant
// instead of having an empty variant.
//
// The main thing we're testing here is that the error message only points to
// the identifier of each variant, not the entire variant itself. Similar to a
// test before, try inserting different spans to the error created and see what
// comes out!
//
//
// Resources
//
//  - The `syn::Error` type
//    https://docs.rs/syn/0.15/syn/struct.Error.html

use sorted::sorted;

use std::env::VarError;
use std::error::Error as StdError;
use std::fmt;
use std::io;
use std::str::Utf8Error;

#[sorted]
pub enum Error {
    Fmt(fmt::Error),
    Io(io::Error),
    Utf8(Utf8Error),
    Var(VarError),
    Dyn(Box<dyn StdError>),
}

fn main() {}
