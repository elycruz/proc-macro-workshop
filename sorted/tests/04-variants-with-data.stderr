error: Dyn should sort before Fmt
   |
15 |     Dyn(Box<dyn StdError>),
   |     ^^^
